src/protocols/protocol_address_31402.cpp

```diff
@@ -102,11 +102,13 @@ bool protocol_address_31402::handle_receive_get_address(const code& ec,
     if (stopped(ec))
         return false;
 
-    // TODO: allowing repeated queries can allow a channel to map our history.
-    // TODO: pull active hosts from host cache (currently just resending self).
-    // TODO: need to distort for privacy, don't send currently-connected peers.
-    // TODO: response size limit is max_address (1000).
+    bc::message::network_address::list addresses;
+    network_.fetch_addresses(addresses);
 
+    static auto generator = std::default_random_engine{pseudo_random()};
+    std::shuffle(addresses.begin(), addresses.end(), generator);
+
+    self_.set_addresses(addresses);
```

Comment:
The reason this was a member was that it was constant (basically the config value). In this new case it can vary on each request.

On the other hand it may be best to allow only one request per connection. However in that case we would not store the previous response (memory wasted) but instead would not re-register the subscription. A true result from this method re-registers and a false deregisters. So just send a local value and return false.

If there is something to send this method needs to call:

```cpp
SEND2(self_, handle_send, _1, self_.command);
```